RealmSwift/Realm.swift

```diff
@@ -1240,6 +1240,18 @@ public final class Realm {
         }
     }
 
+    public func async<C: RangeReplaceableCollectionType where C.Generator.Element: Object>(
```

Comment:
Note that I'm arguing against a `<C: RangeReplaceableCollectionType where C.Generator.Element: Object>` overload, not a `<O: Object>` overload.